File: Sistema.py
from Tienda import Tienda

class Sistema:
    CONTRASEÑA = "123"

    def __init__(self):
        self.tienda = Tienda()

    def ejecutarPrograma(self):
        contraseña_valida = False

        while not contraseña_valida:
            print("*** BIENVENIDO ***")
            contraseña = input("INGRESA LA CONTRASEÑA PARA PODER ACCEDER AL SISTEMA: ")

            if self.validar_contraseña(contraseña):
                contraseña_valida = True
                self.mostrar_menu_sistema()
            else:
                print("Contraseña invalida intenta de nuevo.\n")

    def validar_contraseña(self, contraseña):
        return contraseña == self.CONTRASEÑA

    def mostrar_menu_sistema(self):
        opcion = 0

        while opcion != 11:
            print("ELIGE UNA OPCION PARA CONTINUAR")
            print("*** BIENVENIDOS ***")
            print("1.- Agregar cliente")
            print("2.- Agregar producto")
            print("3.- Añadir a stock")
            print("4.- Disminuir Stock")
            print("5.- Listar clientes")
            print("6.- Consultar productos por categoria")
            print("7.- Listar Productos")
            print("8.- Comprar")
            print("9.- Consultar compras")
            print("10.- Eliminar productos y clientes")
            print("11. Salir")

            opcion = int(input("Ingresa la opcion: "))

            if opcion == 1:
                self.tienda.registrar_cliente()
            elif opcion == 2:
                self.tienda.registrar_producto()
            elif opcion == 3:
                print("Ingresa el tipo de producto al que le deseas añadir stock:")
                print("1.- Limpieza")
                print("2.- Electrodomesticos")
                print("3.- Alimento")
                print("4.- Maquillaje")
                opcion_producto_añadir_stock = int(input("Ingresa la opción: "))

                if opcion_producto_añadir_stock == 1:
                    self.tienda.añadir_stock_limpieza()
                elif opcion_producto_añadir_stock == 2:
                    self.tienda.añadir_stock_electrodomestico()
                elif opcion_producto_añadir_stock == 3:
                    self.tienda.añadir_stock_alimento()
                elif opcion_producto_añadir_stock == 4:
                    self.tienda.añadir_stock_maquillaje()
                else:
                    print("Opción no válida.")
            elif opcion == 4:
                print("Ingresa el tipo de producto al que le deseas eliminar stock:")
                print("1.- Limpieza")
                print("2.- Electrodomesticos")
                print("3.- Alimento")
                print("4.- Maquillaje")
                opcion_producto_eliminar_stock = int(input("Ingresa la opción: "))

                if opcion_producto_eliminar_stock == 1:
                    self.tienda.disminuir_stock_limpieza()
                elif opcion_producto_eliminar_stock == 2:
                    self.tienda.disminuir_stock_electrodomestico()
                elif opcion_producto_eliminar_stock == 3:
                    self.tienda.disminuir_stock_alimento()
                elif opcion_producto_eliminar_stock == 4:
                    self.tienda.disminuir_stock_maquillaje()
                else:
                    print("Opción no válida.")
            elif opcion == 5:
                self.tienda.consultar_clientes()
            elif opcion == 6:
                print("Ingresa el tipo de producto que deseas consultar:")
                print("1.- Limpieza")
                print("2.- Electrodomesticos")
                print("3.- Alimento")
                print("4.- Maquillaje")
                opcion_producto_consulta = int(input("Ingresa la opción: "))

                if opcion_producto_consulta == 1:
                    self.tienda.consultar_productos_limpieza()
                elif opcion_producto_consulta == 2:
                    self.tienda.consultar_productos_electrodomestico()
                elif opcion_producto_consulta == 3:
                    self.tienda.consultar_productos_alimento()
                elif opcion_producto_consulta == 4:
                    self.tienda.consultar_productos_maquillaje()
                else:
                    print("Opción no válida.")
            elif opcion == 7:
                self.tienda.consultar_productos()
            elif opcion == 8:
                self.tienda.realizar_compra()
            elif opcion == 9:
                self.tienda.consultar_compras()
            elif opcion == 10:
                print("ELIMINAR OPCIONES")
                print("1.- Eliminar cliente")
                print("2.- Eliminar producto")
                print("3.- Regresar al menú principal")
                opcionEliminar = int(input("Ingresa la opcion: "))
                if opcionEliminar == 1:
                    self.tienda.eliminar_cliente()
                elif opcionEliminar == 2:
                    self.tienda.eliminar_producto()
                elif opcionEliminar == 3:
                    pass
                else:
                    print("Opción no válida.")
            elif opcion == 11:
                print("Saliendo")
            else:
                print("INGRESE UNA OPCIÓN VALIDA")

File: Sistema.java

import java.util.Scanner;

public class Sistema {
    private final String CONTRASEÑA = "123";
    private Scanner scanner = new Scanner(System.in);
    private Tienda tienda = new Tienda();

    public void ejecutarPrograma() {
        boolean contraseñaValida = false;

        do {
            System.out.println("*** BIENVENIDO ***");
            System.out.print("INGRESA LA CONTRASEÑA PARA PODER ACCEDER AL SISTEMA: ");

            String contraseña = scanner.nextLine();

            if (validarContraseña(contraseña)) {
                contraseñaValida = true;
                mostrarMenuSistema();
            } else {
                System.out.println("Contraseña invalida intenta de nuevo.\n");
            }
        } while (!contraseñaValida);
    }

    private boolean validarContraseña(String contraseña) {
        return contraseña.equals(CONTRASEÑA);

    }

    private void mostrarMenuSistema() {

        int opcion = 0;
        

        do {
            System.out.println("ELIGE UNA OPCION PARA CONTINUAR");
            System.out.println("*** BIENVENIDOS ***");
            System.out.println("1.- Agregar cliente");
            System.out.println("2.- Agregar producto");
            System.out.println("3.- Añadir a stock");
            System.out.println("4.- Disminuir Stock");
            System.out.println("5.- Listar clientes");
            System.out.println("6.- Consultar productos por categoria");
            System.out.println("7.- Listar Productos");
            System.out.println("8.- Comprar");
            System.out.println("9.- Consultar compras");
            System.out.println("10.- Eliminar productos y clientes");
            System.out.println("11. Salir");

            System.out.print("Ingresa la opcion: ");
            opcion = scanner.nextInt();

            switch (opcion) {
                case 1: // Agregar cliente
                    tienda.registrarCliente();

                    break;
                case 2: // Agregar producto
                    System.out.println("Ingresa el tipo de producto que deseas agregar: ");
                    System.out.println("1.- Limpieza");
                    System.out.println("2.- Electrodomesticos");
                    System.out.println("3.- Alimento");
                    System.out.println("4.- Maquillaje");
                    int opcionProducto = scanner.nextInt();

                    switch (opcionProducto) {
                        case 1:
                            tienda.registrarProductoLimpieza();

                            break;
                        case 2:
                            tienda.registrarProductoElectrodomestico();
                            break;

                        case 3:
                            tienda.registrarProductosAlimento();
                            break;

                        case 4:
                            tienda.registrarProductosMaquillaje();
                            break;

                        default:
                            break;
                    }

                    break;

                case 3: // Añadir a stock
                    System.out.println("Ingresa el tipo de producto al que le deseas añadir stock: ");
                    System.out.println("1.- Limpieza");
                    System.out.println("2.- Electrodomesticos");
                    System.out.println("3.- Alimento");
                    System.out.println("4.- Maquillaje");
                    int opcionProductoAñadirStock = scanner.nextInt();

                    switch (opcionProductoAñadirStock) {
                        case 1:
                            tienda.añadirStockLimpieza();
                            break;
                        case 2:
                            tienda.añadirStockElectrodomestico();
                            break;
                        case 3:
                            tienda.añadirStockAlimento();
                            break;
                        case 4:
                            tienda.añadirStockMaquillaje();
                            break;

                    }

                    break;
                case 4: // Disminuir Stock
                    System.out.println("Ingresa el tipo de producto al que le deseas eliminar stock: ");
                    System.out.println("1.- Limpieza");
                    System.out.println("2.- Electrodomesticos");
                    System.out.println("3.- Alimento");
                    System.out.println("4.- Maquillaje");
                    int opcionProductoEliminarStock = scanner.nextInt();

                    switch (opcionProductoEliminarStock) {
                        case 1:
                            tienda.disminuirStockLimpieza();
                            break;
                        case 2:
                            tienda.disminuirStockElectrodomestico();
                            break;
                        case 3:
                            tienda.disminuirStockAlimento();
                            break;
                        case 4:
                            tienda.disminuirStockMaquillaje();
                            break;
                    }

                case 5: // Listar clientes
                    tienda.consultarClientes();
                    break;
                case 6:

                    System.out.println("Ingresa el tipo de producto que deseas consultar");
                    System.out.println("1.- Limpieza");
                    System.out.println("2.- Electrodomesticos ");
                    System.out.println("3.- Alimento ");
                    System.out.println("4.- Maquillaje");

                    int opcionProductoConsulta = scanner.nextInt();

                    switch (opcionProductoConsulta) {
                        case 1:
                            tienda.consultarProductosLimpieza();
                            break;
                        case 2:
                            tienda.consultarProductosElectrodomestico();
                            break;
                        case 3:
                            tienda.consultarProductosCaducidad();
                            break;
                        case 4:
                            tienda.consultarProductosMaquillaje();
                            break;

                    }

                    break;
                case 7:
                    tienda.consultarProductos();
                    break;
                case 8: // Comprar
                    tienda.realizarCompra();
                    break;
                case 9: // Consultar compras
                    tienda.consultarCompras();
                    break;
                case 10: // Eliminar productos y clientes
                    System.out.println("ELIMINAR OPCIONES");
                    System.out.println("1.- Eliminar cliente");
                    System.out.println("2.- Eliminar producto");
                    System.out.println("3.- Regresar al menú principal");
                    int opcionEliminar = scanner.nextInt();

                    switch (opcionEliminar) {
                        case 1:
                            tienda.eliminarCliente();
                            break;
                        case 2:
                            tienda.eliminarProducto();
                            break;
                        case 3:
                            break;
                    }
                    mostrarMenuSistema();
                    return;
                case 11:
                    System.out.println("Saliendo");
                    break;
                default:
                    System.out.println("INGRESE UNA OPCIÓN VALIDA");

            }

        } while (opcion != 11);
    }
}
